folly/experimental/exception_tracer/StackTrace.h

```diff
@@ -85,18 +113,10 @@ class StackTraceStack {
   // In debug mode, we assert that we're in zero-initialized memory by
   // checking that the two guards around top_ are zero.
   void checkGuard() const {
-#ifndef NDEBUG
-    assert(guard1_ == 0 && guard2_ == 0);
-#endif
+    if (kIsDebug) {
+      assert(guard1_ == 0 && guard2_ == 0);
+    }
   }
-
-#ifndef NDEBUG
-  uintptr_t guard1_;
-#endif
-  Node* top_;
```

Comment:
Thinking again ... as a completely different approach to structuring this, and probably way simpler:

```
class StackTraceStack {
 public:
  void checkGuard() const {
    assert(state_[0] == 0 && state_[2] == 0);
  }

 private:
  template <typename Self>
  auto& top(Self& self) {
    return kIsDebug ? self.state_[1] : self.state_[0];
  }

  Node* state_[kIsDebug ? 3 : 1];
};
```

Reply:
Nice. I like this. I implemented effectively this, but I did not see an easy way to run this against the tests in `folly/experimental/exception_tracer`. In my build, `FOLLY_USE_SYMBOLIZER` is defined, so the library gets built, but not the tests from what I can tell. What is the recommended way to run these tests, or it just a matter of adding it to the top-level `CMakeLists.txt`?